Fix: Non-maximized window size too big at startup

david-hall opened the pull request:
This fixes an issue that initially showed up when we switched desktop to Java 17. The main window is correctly maximized at startup, but when it's later switched to normal size (not maximized or minimized) then it was much bigger than the actual screen, at least on Windows 10. I'm not sure about the MAC. The solution is to set the "normal" size after the frame is made visible in the same way as commit 0ff6a4f41ac0d655c203930b5b0a9ba04d215267 did for the maximized window state.

vorth reviewed (commented):
I cannot understand why, but this branch is making vZome crash on the Mac.  I've switched to master and back three times to confirm.  I'm getting a GL error:

```
Caused by: java.lang.RuntimeException: glUseProgram: glError 1281
        at com.vzome.opengl.OpenGlUtilities.checkGLError(OpenGlUtilities.java:59)
        at com.vzome.opengl.SolidRenderer.setLights(SolidRenderer.java:174)
        at org.vorthmann.zome.render.jogl.JoglRenderingViewer.render(JoglRenderingViewer.java:109)
        at org.vorthmann.zome.render.jogl.JoglRenderingViewer.display(JoglRenderingViewer.java:156)
        at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:692)
        at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:674)
        at com.jogamp.opengl.awt.GLCanvas$11.run(GLCanvas.java:1428)
        at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1293)
        ... 15 more
```

david-hall commented:
Can you try running it again after commenting out DocumentFrame line 685 :
// this.setExtendedState(NORMAL);

It works fine on my PC with or without that line so it's obviously not
necessary for the fix I was trying to implement. If that works on the Mac,
I'll remove that line and update the PR.

On Tue, Oct 25, 2022 at 11:48 PM Scott Vorthmann ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> I cannot understand why, but this branch is making vZome crash on the Mac.
> I've switched to master and back three times to confirm. I'm getting a GL
> error:
>
> Caused by: java.lang.RuntimeException: glUseProgram: glError 1281
>         at com.vzome.opengl.OpenGlUtilities.checkGLError(OpenGlUtilities.java:59)
>         at com.vzome.opengl.SolidRenderer.setLights(SolidRenderer.java:174)
>         at org.vorthmann.zome.render.jogl.JoglRenderingViewer.render(JoglRenderingViewer.java:109)
>         at org.vorthmann.zome.render.jogl.JoglRenderingViewer.display(JoglRenderingViewer.java:156)
>         at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:692)
>         at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:674)
>         at com.jogamp.opengl.awt.GLCanvas$11.run(GLCanvas.java:1428)
>         at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1293)
>         ... 15 more
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vZome/vzome/pull/623#pullrequestreview-1155822575>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACTTECWUQQSCURXNJHLIDU3WFCSXDANCNFSM6AAAAAARMUVK64>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

david-hall commented:
Since that line wasn't needed on the PC, I went ahead and removed it and
updated the PR. Please try the new commit with that line removed.

On Wed, Oct 26, 2022 at 12:54 AM David Hall ***@***.***> wrote:

> Can you try running it again after commenting out DocumentFrame line 685 :
> // this.setExtendedState(NORMAL);
>
> It works fine on my PC with or without that line so it's obviously not
> necessary for the fix I was trying to implement. If that works on the Mac,
> I'll remove that line and update the PR.
>
> On Tue, Oct 25, 2022 at 11:48 PM Scott Vorthmann ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>>
>> I cannot understand why, but this branch is making vZome crash on the
>> Mac. I've switched to master and back three times to confirm. I'm getting a
>> GL error:
>>
>> Caused by: java.lang.RuntimeException: glUseProgram: glError 1281
>>         at com.vzome.opengl.OpenGlUtilities.checkGLError(OpenGlUtilities.java:59)
>>         at com.vzome.opengl.SolidRenderer.setLights(SolidRenderer.java:174)
>>         at org.vorthmann.zome.render.jogl.JoglRenderingViewer.render(JoglRenderingViewer.java:109)
>>         at org.vorthmann.zome.render.jogl.JoglRenderingViewer.display(JoglRenderingViewer.java:156)
>>         at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:692)
>>         at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:674)
>>         at com.jogamp.opengl.awt.GLCanvas$11.run(GLCanvas.java:1428)
>>         at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1293)
>>         ... 15 more
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/vZome/vzome/pull/623#pullrequestreview-1155822575>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACTTECWUQQSCURXNJHLIDU3WFCSXDANCNFSM6AAAAAARMUVK64>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
